pkg/mgmt/volume/volume.go

```diff
@@ -49,31 +51,116 @@ func (c *VolController) syncHandler(key string) error {
 	}
 
 	// Get the Vol resource with this namespace/name
-	Vol, err := c.VolLister.LVMVolumes(namespace).Get(name)
+	klog.Infof("Getting lvmvol object name:%s, ns:%s from cache\n", name, namespace)
+	unstructuredVol, err := c.VolLister.Namespace(namespace).Get(name)
 	if k8serror.IsNotFound(err) {
 		runtime.HandleError(fmt.Errorf("lvmvolume '%s' has been deleted", key))
 		return nil
 	}
 	if err != nil {
 		return err
 	}
-	VolCopy := Vol.DeepCopy()
+	vol := &apis.LVMVolume{}
+	err = runtimenew.DefaultUnstructuredConverter.FromUnstructured(unstructuredVol.UnstructuredContent(), &vol)
+	if err != nil {
+		klog.Infof("err %s, While converting unstructured obj to typed object\n", err.Error())
+	}
+	VolCopy := vol.DeepCopy()
 	err = c.syncVol(VolCopy)
 	return err
 }
 
+// addVol is the add event handler for LVMVolume
+func (c *VolController) addVol(obj interface{}) {
+	Vol, ok := c.getStructuredObject(obj)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("Couldn't get Vol object %#v", obj))
+		return
+	}
+
+	if lvm.NodeID != Vol.Spec.OwnerNodeID {
+		return
+	}
+	klog.Infof("Got add event for Vol %s", Vol.Name)
+	klog.Infof("lvmvolume object to be enqueued by Add handler: %v", Vol)
+	c.enqueueVol(Vol)
+}
+
+// updateVol is the update event handler for LVMVolume
+func (c *VolController) updateVol(oldObj, newObj interface{}) {
+	newVol, ok := c.getStructuredObject(newObj)
+	if !ok {
+		runtime.HandleError(fmt.Errorf("Couldn't get Vol object %#v", newVol))
+		return
+	}
+
+	if lvm.NodeID != newVol.Spec.OwnerNodeID {
+		return
+	}
+
+	if c.isDeletionCandidate(newVol) {
+		klog.Infof("Got update event for deleted Vol %s, Deletion timestamp %s", newVol.Name, newVol.ObjectMeta.DeletionTimestamp)
+		c.enqueueVol(newVol)
+	}
+}
+
+// deleteVol is the delete event handler for LVMVolume
+func (c *VolController) deleteVol(obj interface{}) {
+	Vol, ok := c.getStructuredObject(obj)
+	if !ok {
+		unstructuredObj, ok := obj.(*unstructured.Unstructured)
+		if ok {
+			tombStone := cache.DeletedFinalStateUnknown{}
+			err := runtimenew.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.UnstructuredContent(), &tombStone)
+			if err != nil {
+				runtime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
+				return
+			}
+			Vol, ok = tombStone.Obj.(*apis.LVMVolume)
+			if !ok {
+				runtime.HandleError(fmt.Errorf("tombstone contained object that is not a lvmvolume %#v", obj))
+				return
+			}
+		} else {
```

Comment:
if-else is not good practice. Read comment at a previous point with similar pattern.